crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -137,24 +137,20 @@ impl LogManagerActor {
         topic: &Topic,
         mut request: LogUploadCmdPayload,
     ) -> Result<(), ChannelError> {
-        match self
+        if let Err(error) = self
             .execute_logfile_request_operation(topic, &request)
             .await
         {
-            Ok(()) => {
-                request.successful();
-                self.publish_command_status(topic, &request).await?;
-                info!("Log request processed for log type: {}", request.log_type);
-                Ok(())
-            }
-            Err(error) => {
-                let error_message = format!("Handling of operation failed with {}", error);
-                request.failed(&error_message);
-                self.publish_command_status(topic, &request).await?;
-                error!("{}", error_message);
-                Ok(())
-            }
+            let error_message = format!("Handling of operation failed with {}", error);
+            request.failed(&error_message);
+            self.publish_command_status(topic, &request).await?;
+            error!("{}", error_message);
+            return Ok(());
         }
+
+        self.pending_operations.insert(topic.name.clone(), request);
+
+        Ok(())
     }
 
     /// executes the log file request
```

Comment:
The comment from L156-L160 is outdated. Could you also fix it? Actually this function does only creating an upload request and sending it to uploader.

Reply:
Fixed in 2024cd9. New function name is `generate_and_upload_logfile`.